src/dmd/opover.d

```diff
@@ -432,6 +432,43 @@ Expression op_overload(Expression e, Scope* sc, TOK* pop = null)
             }
         }
 
+        override void visit(PostExp pe)
+        {
+            AggregateDeclaration ad = isAggregate(pe.e1.type);
+            if (!ad)
+                return visit(cast(BinExp) pe);
+            auto fdr = search_function(ad, Id.opUnaryRight);
+            if (!fdr)
+                return visit(cast(BinExp) pe);
+
```

Comment:
from here on the function always returns an error. The control flow can be simplified to make that clear.

Reply:
> from here on the function always returns an error

correct.

> The control flow can be simplified to make that clear.

How?